### PR TITLE
test: make tabbed dashboard a little more complex

### PR DESCRIPTION
**Pull request by ktmud**
### SUMMARY

Update the test dashboard "Tabbed dashboard" to have more nested tabs so we can test more complex use cases.

Also updated the `update_slice_ids` function to use dict instead of sorted lists so that slices in position_json can be matched more correctly.

#### Before

Xnip2022-05-05_15-02-12

#### After

Xnip2022-05-05_14-36-59

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/charts/schemas.py [hunk 1]
```diff
@@ -156,7 +156,6 @@ class ChartEntityResponseSchema(Schema):
     slice_name = fields.String(description=slice_name_description)
     cache_timeout = fields.Integer(description=cache_timeout_description)
     changed_on = fields.String(description=changed_on_description)
-    modified = fields.String()
```
Comment (ktmud) [inline superset/charts/schemas.py:159]: By catch, this field returns invalid strings:
Xnip2022-05-05_14-16-49

File: superset/examples/world_bank.py [hunk 2]
```diff
@@ -460,12 +459,6 @@ def create_slices(tbl: BaseDatasource) -> List[Slice]:
         "meta": {"chartId": 49, "height": 50, "sliceName": "Treemap", "width": 8},
         "type": "CHART",
     },
-    "CHART-3nc0d8sk": {
-        "children": [],
-        "id": "CHART-3nc0d8sk",
-        "meta": {"chartId": 50, "height": 50, "sliceName": "Treemap", "width": 8},
-        "type": "CHART",
-    },
```
Comment (ktmud) [inline superset/examples/world_bank.py:468]: Bycatch: this is a duplicate slice where the chart component isn't really used in the layout.

File: superset/examples/helpers.py [hunk 3]
```diff
@@ -39,17 +39,25 @@ def get_examples_folder() -> str:
     return os.path.join(app.config["BASE_DIR"], "examples")
 
 
-def update_slice_ids(layout_dict: Dict[Any, Any], slices: List[Slice]) -> None:
-    charts = [
+def update_slice_ids(pos: Dict[Any, Any]) -> List[Slice]:
```
Comment (john-bodley) [inline superset/examples/helpers.py:42]: Why not return a set if order isn’t relevant?

File: superset/examples/misc_dashboard.py [hunk 4]
```diff
@@ -211,11 +210,7 @@ def load_misc_dashboard() -> None:
     """
     )
     pos = json.loads(js)
-    slices = (
-        db.session.query(Slice).filter(Slice.slice_name.in_(misc_dash_slices)).all()
-    )
-    slices = sorted(slices, key=lambda x: x.id)
-    update_slice_ids(pos, slices)
+    pos, slices = update_slice_ids(pos)
```
Comment (john-bodley) [inline superset/examples/misc_dashboard.py:213]: Shouldn’t this be `slices = …`?

File: superset/models/dashboard.py [hunk 5]
```diff
@@ -218,7 +220,7 @@ def filter_sets_lst(self) -> Dict[int, FilterSet]:
         }
 
     @property
-    def charts(self) -> List[BaseDatasource]:
+    def charts(self) -> List[str]:
```
Comment (john-bodley) [inline superset/models/dashboard.py:223]: How/why wasn’t this caught by Mypy?

File: superset/examples/helpers.py (hunk 3 above)
Comment (ktmud) [reply superset/examples/helpers.py:42]: The type wouldn't match with an SQLA n:n relationship property which expects a list.

File: superset/examples/misc_dashboard.py (hunk 4 above)
Comment (ktmud) [reply superset/examples/misc_dashboard.py:213]: Ah, right! I'm surprised test cases didn't catch this.

File: superset/models/dashboard.py (hunk 5 above)
Comment (ktmud) [reply superset/models/dashboard.py:223]: It was caught when I updated the file! This must be a very old piece of code.

**Comment by ktmud** [comment]
@john-bodley I added an integration test for `superset load_examples` command. Can you take a look?

File: superset/cli/examples.py [hunk 6]
```diff
@@ -54,8 +54,7 @@ def load_examples_run(
     if load_test_data:
         print("Loading [Tabbed dashboard]")
         examples.load_tabbed_dashboard(only_metadata)
-
-    if not load_test_data:
```
Comment (john-bodley) [inline superset/cli/examples.py:58]: Ha!

**Review by john-bodley — APPROVED** [review]

**Pull request merged**